code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -228,7 +233,10 @@
 /obj/machinery/atmospherics/proc/isConnectable(obj/machinery/atmospherics/target, given_layer)
 	if(isnull(given_layer))
 		given_layer = piping_layer
-	if((target.piping_layer == given_layer) || (target.pipe_flags & PIPING_ALL_LAYER))
+	if(
+		((target.piping_layer == given_layer) || (target.pipe_flags & PIPING_ALL_LAYER)) && target.loc != loc \
+		&& (target.pipe_color == pipe_color || target.pipe_flags & PIPING_ALL_COLORS || pipe_flags & PIPING_ALL_COLORS || target.pipe_color == COLOR_VERY_LIGHT_GRAY || pipe_color == COLOR_VERY_LIGHT_GRAY)
+	)
```

Comment:
make this into a proc so it can be written more sane

Reply:
made the if string into 2 procs (one for layer and one for color)